src/batou/component.py

```diff
@@ -349,11 +353,40 @@ def deploy(self, predict_only=False):
                 )
                 output.flush_buffer()
                 if not predict_only:
-                    self.update()
+                    with batou.utils.ComponentTimer(self, "update"):
+                        self.update()
                 self.changed = True
 
         output.clear_buffer()
 
+        verify_duration = self._durations.get("verify", 0)
```

Comment:
and this then maybe can be moved into the timer object (maybe not all of it, but I think most of it).

I guess the timer should not have access to the "output", but it could provide formatting helpers and do the threshold computation etc.